pio-ci: add page (#6186)
tldr-pages/tldr: pages/common/pio-ci.md

# pio ci

> > Build PlatformIO projects with arbitrary source code structure.
> > This will create a new temporary project, which the source code will be copied into.

Reviewer: Perhaps this will read better?
```suggestion
> Build PlatformIO projects with an arbitrary source code structure.
> This will create a new temporary project which the source code will be copied into.
```

By `new temporary project`, do you mean a `new temporary directory`?

Author: It creates a new project within the `/tmp` directory

> More information: <https://docs.platformio.org/en/latest/core/userguide/cmd_ci.html>.

> - Build a PlatformIO project in the temporary directory of the operating system and delete it afterwards:

Reviewer: Perhaps this will read better?
```suggestion
- Build a PlatformIO project in the default system temporary directory and delete it afterwards:
```

`pio ci {{path/to/project}}`

- Build a PlatformIO project and specify specific libraries:

> `pio ci --lib {{path/to/library_directory}} {{path/to/project}}`

Reviewer 2: I was wondering if more than one lib is supported .. something like:
```suggestion
`pio ci --lib {{path/to/library_directory1}} --lib {{path/to/library_directory2}} {{path/to/project}}`
```

Author: You can specify almost everything multiple times, but I think that makes it way harder to read. If someone is interested in the details, we have the link to the documentation in the description.

Reviewer 2: I agree .. You are correct :) .. let's not make it too verbose

- Build a PlatformIO project and specify a specific board (`pio boards` lists all of them):

`pio ci --board {{board}} {{path/to/project}}`

- Build a PlatformIO project in a specific directory:

`pio ci --build-dir {{path/to/build_directory}} {{path/to/project}}`

> - Build a PlatformIO project and do not delete the build directory:

Reviewer 3: ```suggestion
- Build a PlatformIO project and don't delete the build directory:
```

Author: I think that's just a matter of taste. I decided for `do not`, because it's more formal and puts the focus on it, since it's the important part of the description.

Reviewer 3: Sure. 
I _do not_ mind, I just feel it rolls off the tongue a little nasty compared to don't

Author: Maybe @sbrl has an opinion?

Author: Re-reading it a few days later, I actually prefer your suggestion
